services/motion/motion_configuration.go

```diff
@@ -0,0 +1,81 @@
+package motion
+
+import (
+	"math"
+
+	commonpb "go.viam.com/api/common/v1"
+	pb "go.viam.com/api/service/motion/v1"
+
+	"go.viam.com/rdk/protoutils"
+	"go.viam.com/rdk/resource"
+)
+
+func configurationFromProto(motionCfg *pb.MotionConfiguration) *MotionConfiguration {
+	visionSvc := []resource.Name{}
+	planDeviationM := 0.
+	positionPollingHz := 0.
+	obstaclePollingHz := 0.
+	linearMPerSec := 0.
+	angularDegsPerSec := 0.
+
+	if motionCfg != nil {
+		if motionCfg.VisionServices != nil {
+			for _, name := range motionCfg.GetVisionServices() {
+				visionSvc = append(visionSvc, protoutils.ResourceNameFromProto(name))
+			}
+		}
+		if motionCfg.PositionPollingFrequencyHz != nil {
+			positionPollingHz = motionCfg.GetPositionPollingFrequencyHz()
+		}
+		if motionCfg.ObstaclePollingFrequencyHz != nil {
+			obstaclePollingHz = motionCfg.GetObstaclePollingFrequencyHz()
+		}
+		if motionCfg.PlanDeviationM != nil {
+			planDeviationM = motionCfg.GetPlanDeviationM()
+		}
+		if motionCfg.LinearMPerSec != nil {
+			linearMPerSec = motionCfg.GetLinearMPerSec()
+		}
+		if motionCfg.AngularDegsPerSec != nil {
+			angularDegsPerSec = motionCfg.GetAngularDegsPerSec()
+		}
+	}
+
+	return &MotionConfiguration{
+		VisionServices:        visionSvc,
+		PositionPollingFreqHz: positionPollingHz,
+		ObstaclePollingFreqHz: obstaclePollingHz,
+		PlanDeviationMM:       1e3 * planDeviationM,
+		LinearMPerSec:         linearMPerSec,
+		AngularDegsPerSec:     angularDegsPerSec,
+	}
+}
+
+func (motionCfg MotionConfiguration) toProto() *pb.MotionConfiguration {
+	proto := &pb.MotionConfiguration{}
+	if !math.IsNaN(motionCfg.LinearMPerSec) && motionCfg.LinearMPerSec != 0 {
+		proto.LinearMPerSec = &motionCfg.LinearMPerSec
+	}
+	if !math.IsNaN(motionCfg.AngularDegsPerSec) && motionCfg.AngularDegsPerSec != 0 {
+		proto.AngularDegsPerSec = &motionCfg.AngularDegsPerSec
+	}
+	if !math.IsNaN(motionCfg.ObstaclePollingFreqHz) && motionCfg.ObstaclePollingFreqHz > 0 {
+		proto.ObstaclePollingFrequencyHz = &motionCfg.ObstaclePollingFreqHz
+	}
+	if !math.IsNaN(motionCfg.PositionPollingFreqHz) && motionCfg.PositionPollingFreqHz > 0 {
+		proto.PositionPollingFrequencyHz = &motionCfg.PositionPollingFreqHz
+	}
+	if !math.IsNaN(motionCfg.PlanDeviationMM) && motionCfg.PlanDeviationMM >= 0 {
+		planDeviationM := 1e-3 * motionCfg.PlanDeviationMM
+		proto.PlanDeviationM = &planDeviationM
```

Comment:
Feels weird to be taking a pointer for a float, is there a reason this is `&planDeviationM` rather than `planDeviationM`?

Reply:
(same question goes for the additional pointers-to-floats above)

Reply:
I completely agree its weird but this is copy & paste from what existed before: https://github.com/viamrobotics/rdk/pull/2986/files/317503d7c2ff986fe0c5111d8ef4e84d62bc0ad4#diff-e80286b59bcdd1229397e07003b28aa90729c53699d674e1c97332977fd58f09L155-L158

This is a consequence of making the MotionConfiguration proto struct have optional fields: https://github.com/viamrobotics/api/blob/main/proto/viam/service/motion/v1/motion.proto#L131